xchange-ftx/src/main/java/org/knowm/xchange/ftx/service/FtxLendingServiceRaw.java

```diff
@@ -0,0 +1,197 @@
+              exchange.getNonceFactory().createValue(),
+              signatureCreator,
+              subaccount,
+              new FtxSubmitLendingOfferParams(coin, sizeToLend, rate)
+          );
+          list.add(new FtxLendDataDto(coin, op.get().getLocked(), op.get().getOffered(), sizeToLend, rate));
+        } catch (IOException e) {
+          throw new FtxLendingServiceException("Can't lend all for subaccount: " + subaccount + ", coin: " + coin + ", rate: " + rate);
+        }
+      }
+    });
+    return list;
+  }
+
+  public FtxLendDataDto lend(String subaccount, String coin, double size, double rate) {
+    Objects.requireNonNull(coin);
+    if (StringUtils.isNotBlank(coin))
+      throw new FtxLendingServiceException("Coin are blank or empty");
+    if (rate < 0)
+      throw new FtxLendingServiceException("Rate must to be >= 0, subaccount: " + subaccount + ", coin: " + coin);
+    if (size < 0)
+      throw new FtxLendingServiceException("Size must to be >= 0, subaccount: " + subaccount + ", coin: " + coin + ", rate: " + rate);
+
+    try {
+      FtxLendingInfoDto info = info(subaccount, coin);
+      double sizeToLend = FtxAdapters.lendingRounding(new BigDecimal(size)).doubleValue();
+
+      if (Double.compare(sizeToLend, info.getLendable()) == 1) {
+        throw new FtxLendingServiceException("Can't lend sizeToLend > to lendable, subaccount: " + subaccount + ", coin: " + coin + ", size: " + size + ", sizeToLend: " + sizeToLend + ", rate: " + rate);
+      }
+      ftx.submitLendingOffer(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount,
+          new FtxSubmitLendingOfferParams(coin, FtxAdapters.lendingRounding(new BigDecimal(sizeToLend)).doubleValue(), rate)
+      );
+      return new FtxLendDataDto(coin, info.getLocked(), info.getOffered(), sizeToLend, rate);
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't lend subaccount: " + subaccount + ", coin: " + coin + ", size: " + size + ", rate: " + rate, e);
+    }
+  }
+
+  public List<FtxLendingHistoryDto> histories(String subaccount) {
+    try {
+      return ftx.getlendingHistories(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount
+      ).getResult();
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't get lending infos subAccount: " + subaccount, e);
+    }
+  }
+
+  public List<FtxLendingHistoryDto> histories(String subaccount, List<String> coins) {
+    Objects.requireNonNull(coins);
+    return histories(subaccount).stream()
+        .filter(lendingHistory -> coins.contains(lendingHistory.getCoin()))
+        .collect(Collectors.toList());
+  }
+
+  public FtxLendingHistoryDto history(String subaccount, String coin) {
```

Comment:
same thing as my comment with rates just eliminate this method

Reply:
same comment